strawberryfields/utils.py

```diff
@@ -736,6 +736,7 @@ def is_channel(prog):
     Returns:
         bool: True if all operations in the program are of types :class:`strawberryfields.ops.Gate` and :class:`strawberryfields.ops.Channel`
     """
+    # FIXME isn't a preparation also a quantum channel?
```

Comment:
I'd say no. A channel has an arbitrary input, and gives some related output, whereas a preparation doesn't have any "input" slot.

Reply:
Technically a quantum channel linearly maps valid states to valid states. A preparation maps any state to a fixed state, so you can represent it as a quantum channel  \rho \mapsto \trace(\rho) \xi. But this is a technicality, it all depends on what `is_channel` is meant to do.